src/ARC/ARC.fs

```diff
@@ -94,6 +94,64 @@ type ARC =
 
     // to-do: function that returns read contracts based on a list of paths.
     // the list of paths is used to create a filesystem tree
-    static member createReadContracts (paths : string array) : Contract array =
-        let fileTree = FileSystemTree.fromFilePaths paths
-        raise (System.NotImplementedException())
\ No newline at end of file
+    static member createReadContracts (filePaths : string array) : Contract array =
+        let xlsxReadContractFromPath (path: string) = Contract.createRead(path, DTOType.Spreadsheet)
+        let fs = FileSystem.fromFilePaths filePaths
+        let xlsxFileNames = [|Path.AssayFileName; Path.StudyFileName; Path.InvestigationFileName|]
+        let xlsxFiles = fs.Tree.Filter(fun p -> xlsxFileNames |> Array.contains p)
+        match xlsxFiles with
+        | Some xlsxPaths -> xlsxPaths.ToFilePaths() |> Array.map xlsxReadContractFromPath
+        | None -> [||]
+
+    [<NamedParams(fromIndex=1)>]
+    static member createFromReadContracts (cArr: Contract [], ?enableLogging: bool) =
+        let enableLogging = defaultArg enableLogging false
+        /// filter to only keep *fullfilled* READ contracts of type spreadsheet.
+        let filteredContracts = cArr |> Array.choose (fun c ->
+            match c with
+            | {Operation = READ; DTOType = Some DTOType.Spreadsheet; DTO = Some (DTO.Spreadsheet fsworkbook); Path = p} ->
+                Some (p, fsworkbook)
+            | _ -> None
+        )
+        /// get investigation from xlsx
+        let investigation = 
+            filteredContracts 
+            |> Array.find (fun c -> fst c |> FileSystem.Path.isFile Path.InvestigationFileName)
+            |> snd 
+            |> ISA.Spreadsheet.ArcInvestigation.fromFsWorkbook
+        /// get studies from xlsx
+        let studies =
+            filteredContracts 
+            |> Array.filter (fun c -> fst c |> FileSystem.Path.isFile Path.StudyFileName)
+            |> Array.map (snd >> ISA.Spreadsheet.ArcStudy.fromFsWorkbook)
+        /// get assays from xlsx
+        let assays =
+            filteredContracts 
+            |> Array.filter (fun c -> fst c |> FileSystem.Path.isFile Path.AssayFileName)
+            |> Array.map (snd >> ISA.Spreadsheet.ArcAssay.fromFsWorkbook)
+        /// Create a investigation copy to check for registered studies/assays
+        let copy = investigation.Copy()
+        /// Get all registered studies
+        let registeredStudies = copy.StudyIdentifiers
+        registeredStudies |> Seq.iter (fun studyRegisteredIdent ->
+            /// Try find registered study in parsed READ contracts
+            let studyOpt = studies |> Array.tryFind (fun s -> s.Identifier = studyRegisteredIdent)
```

Comment:
Yeah if i don't do this, i more or less need to implement the whole "Identifier" logic from ISA.Spreadsheet again. I just parse everything and check identifiers afterwards. This way we have this logic maintainable.

Otherwise this was my idea for study:

```fsharp
let expectedStudyPathFromStudyIdentifier (studyIdentifier) =
    FileSystem.Path.combineMany[|Path.ARCtrl.StudiesFolderName; studyIdentifier; Path.ARCtrl.StudyFileName|]

let tryFindStudy (contracts: Contract []) (studyRegisteredIdent: string) =
    contracts |> Array.tryPick (fun c ->
        let expectedPath = expectedStudyPathFromStudyIdentifier studyRegisteredIdent
        match c with
        | {Operation = READ; DTOType = Some DTOType.ISA_Study; DTO = Some (DTO.Spreadsheet fsworkbook); Path = p} when p = expectedPath ->
            Some (p, fsworkbook)
        | _ -> None
    )
```

But if identifier and foldername are not the same anymore this will simply not work anymore.